frappe/public/js/frappe/model/perm.js

```diff
@@ -117,20 +119,38 @@ $.extend(frappe.perm, {
 	},
 
 	get_role_permissions: (meta) => {
+		/** Returns a `dict` of evaluated Role Permissions like:
+		{
+			"read": 1,
+			"write": 0,
+			"if_owner": [if "if_owner" is enabled]
+				{
+					"read": 1,
+					"write": 0
+				}
+		} */
 		let perm = [{ read: 0, permlevel: 0 }];
-		// Returns a `dict` of evaluated Role Permissions
+
 		(meta.permissions || []).forEach((p) => {
-			// if user has this role
 			let permlevel = cint(p.permlevel);
 			if (!perm[permlevel]) {
 				perm[permlevel] = {};
 				perm[permlevel]["permlevel"] = permlevel;
 			}
 
+			// if user has this role
 			if (frappe.user_roles.includes(p.role)) {
 				frappe.perm.rights.forEach((right) => {
 					let value = perm[permlevel][right] || p[right] || 0;
-					if (value) {
+
+					if (p.if_owner && value) {
+						// if_owner is enabled for perm,
+						// construct perm object inside "if_owner" property
+						if (!perm[permlevel]["if_owner"]) {
+							perm[permlevel]["if_owner"] = {}
+						}
+						perm[permlevel]["if_owner"][right] = value;
```

Comment:
This won't work because `if_owner` should only be applied if all permissions for this right have `if_owner` checked. In other words, if any permission for this right doesn't have `if_owner` checked, `if_owner` shouldn't apply.

Also, `if_owner` is only required for permlevel 0.

Fixing in new commit.